overlord/snapstate/refreshhints.go

```diff
@@ -71,12 +73,24 @@ func (r *refreshHints) refresh() error {
 	perfTimings := timings.New(map[string]string{"ensure": "refresh-hints"})
 	defer perfTimings.Save(r.state)
 
+	var updates []*snap.Info
+	var ignoreValidationByInstanceName map[string]bool
 	timings.Run(perfTimings, "refresh-candidates", "query store for refresh candidates", func(tm timings.Measurer) {
-		_, _, _, err = refreshCandidates(auth.EnsureContextTODO(), r.state, nil, nil, &store.RefreshOptions{RefreshManaged: refreshManaged})
+		updates, _, ignoreValidationByInstanceName, err = refreshCandidates(auth.EnsureContextTODO(), r.state, nil, nil, &store.RefreshOptions{RefreshManaged: refreshManaged})
 	})
 	// TODO: we currently set last-refresh-hints even when there was an
 	// error. In the future we may retry with a backoff.
 	r.state.Set("last-refresh-hints", time.Now())
+	if err == nil {
+		deviceCtx, err := DevicePastSeeding(r.state, nil)
```

Comment:
do we need to use DevicePastSeeding? this is guarded by CanAutoRefresh so seeding is done? is the problem remodeling?  if there's a remodeling in progress we are probably fine, we just need to remember to clear the refresh-hints when is committed?  anyway if really need DevicePastSeeding we should ignore ConflictErrors and return nil for them?

Reply:
Ah, no, no deep reason for this, I rather mindlessly copied this from other code where device context was needed where in fact `DeviceCtx(st, nil, nil)` can be used. Thanks